openreview/venue/venue.py

```diff
@@ -1009,6 +1015,125 @@ def set_SAC_ethics_review_process(self, sac_ethics_flag_duedate=None):
+                group_type="ASSIGNMENT",
+            )
+            iThenticate_submission_id = res["id"]
+            
+
+            # Create a new edge with head as the note ID and tail as the iThenticate submission ID
+            iThenticate_edge = openreview.api.Edge(
+                invitation=self.get_iThenticate_plagiarism_check_invitation_id(),
+                head=submission.id,
+                tail=iThenticate_submission_id,
+                label="Created",
+                weight=-1,
+            )
+
+            
+            iThenticate_edge = self.client.post_edge(iThenticate_edge)
+
+            # Get the PDF for the submission
+            submission_file_binary_data = self.client.get_attachment(
+                id=submission.id, field_name="pdf"
+            )
+
+            submission_file_object = io.BytesIO(submission_file_binary_data)
+
+
+            # update the edge with status file sent
+            iThenticate_edge.label = "File Sent"
+            iThenticate_edge = self.client.post_edge(iThenticate_edge)
+
+
+            # Upload PDF to iThenticate
+            res = iThenticate_client.upload_submission(
+                submission_id=iThenticate_submission_id,
+                file_data=submission_file_object,
+                file_name=submission.content['title']['value']
+            )
+            
+
+    def ithenticate_request_similarity_report(self):
+        if not self.iThenticatePlagiarismCheck:
+            raise openreview.OpenReviewException('iThenticatePlagiarismCheck is not enabled for this venue.')
+        
+        iThenticate_client = openreview.api.iThenticateClient(self.iThenticatePlagiarismCheckApiKey, self.iThenticatePlagiarismCheckApiBaseUrl)
+
+        edges = self.client.get_grouped_edges(
+            invitation=self.get_iThenticate_plagiarism_check_invitation_id(), groupby='tail'
+        )
+
+        for edge in edges:
+            # change edge status to similarity requested
+            if edge["values"][0].label == "File Uploaded":
+                edge["values"][0].label = "Similarity Requested"
+                updated_edge = self.client.post_edge(edge["values"][0])
+
+                iThenticate_client.generate_similarity_report(
+                    submission_id=updated_edge.tail,
+                    search_repositories=[
+                        "INTERNET",
+                        "SUBMITTED_WORK",
+                        "PUBLICATION",
+                        "CROSSREF",
+                        "CROSSREF_POSTED_CONTENT",
+                    ],
+                )
+
+    def check_ithenticate_status(self, type):
+        if not self.iThenticatePlagiarismCheck:
+            raise openreview.OpenReviewException('iThenticatePlagiarismCheck is not enabled for this venue.')
+
+        edges = self.client.get_grouped_edges(
+            invitation=self.get_iThenticate_plagiarism_check_invitation_id(), groupby='tail'
+        )
+        if type == 'Upload':
```

Comment:
can the type match with the label so we can get different statuses?

Reply:
I've added that change